src/lib/actions/diff-create-action.ts

```
@@ -0,0 +1,38 @@
+import {TelemetryService} from '../core-extension-service';
+import * as Constants from '../constants';
+
+export type DiffCreateCallback = (code: string, file?: string) => Promise<void>;
+
+export type DiffCreateDependencies = {
+	callback: DiffCreateCallback;
+	telemetryService: TelemetryService;
+};
+
+export class DiffCreateAction {
+	private readonly commandName: string;
+	private readonly callback: DiffCreateCallback;
+	private readonly telemetryService: TelemetryService;
+
+	public constructor(commandName: string, dependencies: DiffCreateDependencies) {
+		this.commandName = commandName;
+		this.callback = dependencies.callback;
+		this.telemetryService = dependencies.telemetryService;
+	}
+
+	public async run(code: string, file?: string): Promise<void> {
+		const startTime = Date.now();
+		try {
+			await this.callback(code, file);
+			this.telemetryService.sendCommandEvent(Constants.TELEM_DIFF_SUGGESTION, {
+				commandName: this.commandName,
+				commandSource: 'PLACEHOLDER', // NOT SURE WHAT SHOULD GO HERE.
+			});
+		} catch (e) {
+			const errMsg = e instanceof Error ? e.message : e as string;
+			this.telemetryService.sendException(Constants.TELEM_DIFF_SUGGESTION_FAILED, errMsg, {
+				executedCommand: this.commandName,
+				duration: (Date.now() - startTime).toString()
+			});
```

Out of curiosity, what does this do? Where does the exception go? 
And do we want to swallow this exception or rethrow it?

---

It sends a telemetry event indicating that we got an exception. I'm not entirely sure where they go, but we've already been doing this style of telemetry for when SFCA fails.
That said, you're almost certainly right that we should be rethrowing instead of swallowing, though in practice it's probably not terribly relevant because the accept/reject functionality should be pretty stable.